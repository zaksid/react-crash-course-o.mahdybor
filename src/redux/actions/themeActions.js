import Constants from '../constants';

export const setResultsTheme = theme => ({
    type: Constants.SetResultsTheme,
    payload: theme
});

export const setToolbarTheme = theme => ({
    type: Constants.SetToolbarTheme,
    payload: theme
});
