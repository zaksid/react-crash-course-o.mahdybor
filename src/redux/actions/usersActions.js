import Constants from '../constants';

export const fetchUsersStart = () => ({
    type: Constants.FetchUserStart
});

export const usersLoaded = users => ({
    type: Constants.FetchUserSuccess,
    payload: users
});

export const fetchCancel = () => ({
    type: Constants.FetchUserCancel
});

export const fetchError = error => ({
    type: Constants.FetchUserError,
    payload: error
});
