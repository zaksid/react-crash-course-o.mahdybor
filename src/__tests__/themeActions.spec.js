import { setResultsTheme, setToolbarTheme } from '../redux/actions/themeActions';

describe('themeActions', () => {
    describe('setResultsTheme', () => {
        const action = setResultsTheme('dark');

        it('should return correct action type', () => {
            expect(action.type).toBe('SET_RESULTS_THEME');
        });

        it('should match snapshot', () => {
            expect(action).toMatchSnapshot();
        });
    });

    describe('setToolbarTheme', () => {
        const action = setToolbarTheme('light');

        it('should return correct action type', () => {
            expect(action.type).toBe('SET_TOOLBAR_THEME');
        });

        it('setToolbarTheme should match snapshot', () => {
            expect(action).toMatchSnapshot();
        });
    });
});
