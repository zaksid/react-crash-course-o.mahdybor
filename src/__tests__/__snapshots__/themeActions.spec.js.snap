// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeActions setResultsTheme should match snapshot 1`] = `
Object {
  "payload": "dark",
  "type": "SET_RESULTS_THEME",
}
`;

exports[`themeActions setToolbarTheme setToolbarTheme should match snapshot 1`] = `
Object {
  "payload": "light",
  "type": "SET_TOOLBAR_THEME",
}
`;
